src/commands/coverage/upload.ts

```diff
@@ -0,0 +1,356 @@
+    }
+
+    const payloads: Payload[] = []
+    if (reports.length) {
+      for (let i = 0; i < reports.length; i++) {
+        const {format, paths} = reports[i]
+        for (let j = 0; j < paths.length; j += MAX_REPORTS_PER_REQUEST) {
+          const chunkedPaths = paths.slice(j, j + MAX_REPORTS_PER_REQUEST)
+          payloads.push({
+            format,
+            paths: chunkedPaths,
+            // send flush signal with the last payload, and only if the flush flag is set
+            flush: i === reports.length - 1 && j + MAX_REPORTS_PER_REQUEST >= paths.length ? !!this.flush : false,
+            spanTags,
+            customTags,
+            customMeasures,
+            hostname: os.hostname(),
+          })
+        }
+      }
+    } else if (this.flush) {
+      // no reports to upload, only send the flush signal
+      payloads.push({
+        format: undefined,
+        paths: [],
+        flush: true,
+        spanTags,
+        customTags,
+        customMeasures,
+        hostname: os.hostname(),
+      })
+    }
+
+    return payloads
+  }
+
+  private async getSpanTags(): Promise<SpanTags> {
+    const ciSpanTags = getCISpanTags()
+    const gitSpanTags = await getGitMetadata()
+    const userGitSpanTags = getUserGitSpanTags()
+
+    const spanTags = {
+      ...gitSpanTags,
+      ...ciSpanTags,
+      ...userGitSpanTags,
+      ...(this.config.env ? {env: this.config.env} : {}),
+    }
+
+    if (!spanTags[GIT_REPOSITORY_URL]) {
+      throw new Error('git repository URL is missing')
+    }
+
+    if (!spanTags[GIT_SHA]) {
+      throw new Error('git commit SHA is missing')
+    }
+
+    return spanTags
+  }
+
+  private getCustomTags(): Record<string, string> {
+    const envVarTags = this.config.envVarTags ? parseTags(this.config.envVarTags.split(',')) : {}
+    const cliTags = this.tags ? parseTags(this.tags) : {}
+
+    return {
+      ...cliTags,
+      ...envVarTags,
+    }
+  }
+
+  private getCustomMeasures(): Record<string, number> {
+    const envVarMeasures = this.config.envVarMeasures ? parseMetrics(this.config.envVarMeasures.split(',')) : {}
+    const cliMeasures = this.measures ? parseMetrics(this.measures) : {}
+
+    return {
+      ...cliMeasures,
+      ...envVarMeasures,
+    }
+  }
+
+  private async getMatchingCoverageReportFilesByFormat(): Promise<{format: string; paths: string[]}[]> {
+    const codeCoverageReportFiles = (this.basePaths || [])
+      .reduce((acc: string[], basePath: string) => {
+        if (isFile(basePath)) {
+          return acc.concat(fs.existsSync(basePath) ? [basePath] : [])
+        }
+        let globPattern
+        // It's either a folder or a glob pattern
+        if (glob.hasMagic(basePath)) {
+          // It's a glob pattern so we just use it as is
+          globPattern = basePath
+        } else {
+          // It's a folder
+          globPattern = buildPath(basePath, '*.xml') // TODO update the logic to not assume that every report has .xml extension
```

Comment:
What extension could reports have otherwise? If the current implementation only supports XML files, I don't think the todo comment is necessary

Reply:
In the future they can be `.json` or some other format. I agree that the comment is redundant though, removed it